### PR TITLE
Tutorial on how to mint an NFT

### PR DESCRIPTION
**Pull request by smudgil**
This tutorial is part 2/3 of a series on minting an NFT.

This tutorial describes how to interact with an smart contract to mint an NFT using Web3.

### CODE REVIEW

File: src/content/developers/tutorials/how-to-mint-an-nft.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
+
+# How to Mint an NFT (Part 2/3 of NFT Tutorial Series)
+
+This tutorial describes how to mint an NFT on the Ethereum blockchain using our smart contract and Web3. You should have already completed Part 1: How to Write & Deploy an NFT, prior to starting Part 2 below. In Part 3, we will explain how to view your freshly minted NFT on Metamask!
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-mint-an-nft.md:4]: Would add link here for `Part 1: How to Write & Deploy an NFT` to go to part 1 tutorial. Same with `Part 3`

File: src/content/developers/tutorials/how-to-mint-an-nft.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+
+##
+
+[Beeple](https://www.wsj.com/articles/beeple-nft-fetches-record-breaking-69-million-in-christies-sale-11615477732): $69 Million
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-mint-an-nft.md:8]: WSJ has a paywall, is there any link we could use that doesn't include a paywall?

File: src/content/developers/tutorials/how-to-mint-an-nft.md [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+[3LAU](https://www.forbes.com/sites/abrambrown/2021/03/03/3lau-nft-nonfungible-tokens-justin-blau/?sh=5f72ef64643b): $11 Million
+[Grimes](https://www.theguardian.com/music/2021/mar/02/grimes-sells-digital-art-collection-non-fungible-tokens): $6 Million
+
+All of them minted their NFT’s using Alchemy’s powerful API. In this tutorial, we’ll teach you how to the same in <10 minutes.
+
+“Minting a NFT” is the act of publishing a unique instance of your ERC721 token on the blockchain. Using our smart contract from Part 1 of this NFT tutorial series, let’s flex our web3 skills and mint a NFT. At the end of this tutorial, you’ll be able to mint as many NFTs as your heart (and wallet) desires!
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-mint-an-nft.md:14]: Similar note to before about hyphenating ERC's for site convention, ie `ERC-721`

File: src/content/developers/tutorials/how-to-mint-an-nft.md [hunk 4]
```diff
@@ -0,0 +1,275 @@
+
+# How to Mint an NFT (Part 2/3 of NFT Tutorial Series)
+
+This tutorial describes how to mint an NFT on the Ethereum blockchain using our smart contract and Web3. You should have already completed Part 1: How to Write & Deploy an NFT, prior to starting Part 2 below. In Part 3, we will explain how to view your freshly minted NFT on Metamask!
+
+##
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-mint-an-nft.md:6]: would remove this (L6)

Also, reminder to make sure custom ID headers are used throughout if the line starts with hash marks
ie:
`## Section name {#section-name}`

File: src/content/developers/tutorials/how-to-mint-an-nft.md [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+
+Let’s get started! 🪄
+
+## Step 1: Install web3
+
+If you followed the first tutorial on creating your NFT smart contract, you already have experience using Ethers.js. Web3 is similar to Ethers, as it is a library used to make creating requests to the Ethereum blockchain easier. In this tutorial we’ll be using [Alchemy Web3](https://docs.alchemyapi.io/alchemy/documentation/alchemy-web3), which is an enhanced web3 library that offers automatic retries and robust WebSocket support.
+
+In your project home directory run:
+
+    npm install @alch/alchemy-web3
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-mint-an-nft.md:24]: reminder that this site requires backticks to denote inline code, and triple backticks for code blocks, not just indentation (please update throughout)

File: src/content/developers/tutorials/how-to-mint-an-nft.md [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+
+* Navigate to the “Pinata Upload” button on the top right
+
+* Upload an image to pinata — this will be the image asset for your NFT. Feel free to name the asset whatever you wish
+
+* After you upload, at the top of the page, there should be a green popup that allows you to view the hash of your upload → Copy that hashcode. You can view your upload at: [https://gateway.pinata.cloud/ipfs/<](https://gateway.pinata.cloud/ipfs/QmarPqdEuzh5RsWpyH2hZ3qSXBCzC5RyK3ZHnFkAsk7u2f)hash-code>
+
+For the more visual learners, the steps above are summarized here:
+
+![How to upload your image to Pinata](https://cdn-images-1.medium.com/max/2000/0*V_-3PGxnfsnLTGo4.gif)
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-mint-an-nft.md:66]: (reminder to download external images and place into same folder as markdown file, then reference locally using `alt text for gif`)

File: src/content/developers/tutorials/how-to-mint-an-nft.md [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+
+    API_URL = "https://eth-ropsten.alchemyapi.io/v2/your-api-key"
+    PRIVATE_KEY = "your-private-account-address"
+    PUBLIC_KEY = "your-public-account-address"
+
+## Step 7: Create your transaction
+
+First, let’s define a function called mintNFT(tokenData) and create our transaction by doing the following:
+
+1. Grab your PRIVATE_KEY* *and* *PUBLIC_KEY from the .env file.
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-mint-an-nft.md:119]: This renders
image
Was that intentional?
Typically we're using single underscore wrapping to denote italics, and double asterisk wrapping to denote bold, which must touch the content they're wrapping without a space.

File: src/content/developers/tutorials/how-to-mint-an-nft.md [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+
+1. Next, we’ll need to figure out the account nonce. The nonce specification is used to keep track of the number of transactions sent from your address— which we need for security purposes and to prevent [replay attacks](https://docs.alchemyapi.io/alchemy/resources/blockchain-glossary#account-nonce). To get the number of transactions sent from your address, we use [getTransactionCount](https://docs.alchemyapi.io/alchemy/documentation/alchemy-api-reference/json-rpc#eth_gettransactioncount).
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-mint-an-nft.md:121]: We have local reference material for both of these external links. Can we change these to:
https://ethereum.org/en/glossary/#nonce
https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_gettransactioncount

File: src/content/developers/tutorials/how-to-mint-an-nft.md [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+
+    The hash of your transaction is: 0x10e5062309de0cd0be7edc92e8dbab191aa2791111c44274483fa766039e0e00
+
+    Check Alchemy's Mempool to view the status of your transaction!
+
+Next, visit your [Alchemy mempool](https://dashboard.alchemyapi.io/mempool) to see the status of your transaction (whether it’s pending, mined, or got dropped by the network). If your transaction got dropped, it’s also helpful to check [Ropsten Etherscan](https://ropsten.etherscan.io/) and search for your transaction hash.
+
+![View your NFT transaction hash on Etherscan](https://cdn-images-1.medium.com/max/5200/0*Q7IfLIhtVnNtAJn-.png)*View your NFT transaction hash on Etherscan*
+
+And that’s it! You’ve now deployed AND minted with a NFT on the Ethereum blockchain 🤑
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-mint-an-nft.md:271]: We use an `Emoji` component to standardize appearance of all Emoji's.. Just change this line to 
`And that’s it! You’ve now deployed AND minted with a NFT on the Ethereum blockchain <Emoji text=":money_mouth_face:" size={1} />`

File: src/content/developers/tutorials/how-to-mint-an-nft.md [hunk 10]
```diff
@@ -0,0 +1,275 @@
...
+
+Using the mint-nft.js you can mint as many NFT's as your heart (and wallet) desires! Just be sure to pass in a new tokenURI describing the NFT's metadata (otherwise, you'll just end up making a bunch of identical ones with different IDs).
+
+Presumably, you’d like to be able to show off your NFT in your wallet — so be sure to check out Part 3: How to View Your NFT in Your Wallet!
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-mint-an-nft.md:275]: Link this to 3rd part of tutorial (don't worry that it's in another PR)

**Review by wackerow — CHANGES_REQUESTED** [review]
Left more comments here, see review of #2652 first. Let us know if you have questions =)

**Review by Finn-Developments — APPROVED** [review]

**Comment by wackerow** [comment]
@smudgil Hey! Just wanted to reach out and see if you intended on moving forward with this. Feel free to ask us any questions if you have them (here or on Discord)

**Comment by smudgil** [comment]
Hey Paul! Thanks for following up! I do intend to move forward with this, just been super slammed at the moment. I'll get it done by early next week. Thanks a ton for your feedback 🙂
[https://my-email-signature.link/signature.gif?u=1100946&e=147596515&v=efcf5fc7612632f172580ac1c4ad3575550a04b239ef71a342268ad9baf9b047]
________________________________

**Review by smudgil — COMMENTED** [review]
updated files, would appreciate review!

**Comment by smudgil** [comment]
Hey Paul! Pushed some edits near the end of last week, would appreciate feedback! Thanks 🙂
[https://my-email-signature.link/signature.gif?u=1100946&e=149622965&v=363a1ec6340dce7b48c2a22b72a5ddeb1c6577c1e88f95f615d9873f13799b0b]
________________________________

File: src/content/developers/tutorials/how-to-mint-an-nft/index.md [hunk 11]
```diff
@@ -1,29 +1,33 @@
...
-# How to Mint an NFT (Part 2/3 of NFT Tutorial Series)
-
-This tutorial describes how to mint an NFT on the Ethereum blockchain using our smart contract and Web3. You should have already completed Part 1: How to Write & Deploy an NFT, prior to starting Part 2 below. In Part 3, we will explain how to view your freshly minted NFT on Metamask!
-
-##
-
-[Beeple](https://www.wsj.com/articles/beeple-nft-fetches-record-breaking-69-million-in-christies-sale-11615477732): $69 Million
+---
+title: How to Mint an NFT (Part 2/3 of NFT Tutorial Series)
+description: This tutorial describes how to mint an NFT on the Ethereum blockchain using our smart contract and Web3. You should have already completed [Part 1: How to Write & Deploy an NFT](/developers/tutorials/how-to-write-and-deploy-an-nft), prior to starting Part 2 below. In [Part 3, we will explain how to view your freshly minted NFT on Metamask](/developers/tutorials/how-to-view-nft-in-metamask/)!
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-mint-an-nft/index.md:3]: Same as other two, trim down (metadata, not rendered)

File: src/content/developers/tutorials/how-to-mint-an-nft/index.md [hunk 12]
```diff
@@ -32,7 +36,7 @@ Inside your scripts directory, create an mint-nft.js file and add the following
     const { createAlchemyWeb3 } = require("@alch/alchemy-web3");
     const web3 = createAlchemyWeb3(API_URL);
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-mint-an-nft/index.md]: Note, though the indented text will render as code, but we can take advantage of the syntax highlighting if we wrap these code blocks in triple-back-ticks as you've done in the others, put the programming language immediately after the opening back-ticks.  (same for code blocks below)

File: src/content/developers/tutorials/how-to-mint-an-nft/index.md [hunk 13]
```diff
@@ -134,6 +138,7 @@ First, let’s define a function called mintNFT(tokenData) and create our transa
 
 Your mint-nft.js file should look like this now:
 
+ ```js
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-mint-an-nft/index.md:141]: ah yes =) We should extend this formatting to the other blocks of code

**Review by wackerow — CHANGES_REQUESTED** [review]
@smudgil Appreciate the work you're doing on this, and we're getting super close! Few other things to clean up on the three PRs, looking forward to getting this merged =)

**Review by minimalsm — APPROVED** [review]

**Comment by smudgil** [comment]
Hey! Circling back a year later - though about a different tutorial - specifically the NFT Minter Tutorial<https://ethereum.org/en/developers/tutorials/nft-minter/>, which @nstrike2 claims to have authored on the Ethereum org's website and now on Twitter<https://ethereum.org/en/developers/tutorials/nft-minter/>.

I wrote this tutorial over a year ago while I was an interning at Alchemy. You can read it here on web3university<https://www.web3.university/article/nft-minter-tutorial-how-to-create-a-full-stack-dapp> and on the alchemy docs<https://docs.alchemy.com/alchemy/tutorials/nft-minter>. A quick look at the history of the git repo mentioned in the tutorial<https://github.com/alchemyplatform/nft-minter-tutorial/graphs/contributors> also illustrates that I wrote the code for it.

I would really appreciate it if someone could help me take credit for what's rightfully mine. Thank you so much and hope to hear from you soon.

Best regards,
Sumi Mudgil

________________________________

**Comment by nstrike2** [comment]
Hey @paulrberg, following up on this as well. My job while interning at Alchemy was just to put up the existing NFT Minter tutorial from our website on the Ethereum Foundation. After looking at the thread of comments a few months later on this PR, @smudgil is absolutely right — she is the rightful owner of this tutorial, and I had no clue!

I've gone ahead and taken down the Twitter post that Sumi linked, and in addition, I made a PR [here](https://github.com/ethereum/ethereum-org-website/pull/5990) with a request to change the top level name to Sumi's GitHub. Please see that this PR is merged, and Sumi becomes the official rightful owner here.

Once again, my sincerest apologies for my oversight here.

Thank you!!

**Pull request merged**